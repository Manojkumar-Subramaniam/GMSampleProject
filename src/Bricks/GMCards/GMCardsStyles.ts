import {StyleSheet} from 'react-native';
import {Colors} from '../../Utils';
import colors from '../../Utils/colors';

const styles = StyleSheet.create({
  rootView: {
    zIndex: 1,
    elevation: 1,
    shadowRadius: 5,
    margin: 5,
    shadowOpacity: 0.25,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    borderRadius: 10,
    shadowColor: Colors.GMCARD_SHADOW,
    backgroundColor: colors.GMCARD_PRIMARY,
  },
});

export default styles;
