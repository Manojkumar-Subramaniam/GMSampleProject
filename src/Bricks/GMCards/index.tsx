import React from 'react'
import { View, ViewProps } from 'react-native'
import styles from './GMCardsStyles'

interface Props {
    children?: React.ReactNode
    customStyle?: ViewProps
}

const GMCards: React.FC<Props> = ({ children, customStyle }) => {
    return (
        <View style={[styles.rootView, customStyle]}>
            {children}
        </View>
    )
}

export default GMCards