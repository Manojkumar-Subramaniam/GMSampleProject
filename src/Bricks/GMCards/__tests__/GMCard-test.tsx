import React from 'react';
import renderer from 'react-test-renderer';
import GMCards from '..';

describe('GMCard Tests', () => {
  test('GMCard Snapshot', () => {
    const GMCardSnapshot = renderer.create(<GMCards onPress={() => { }} />).toJSON();
    expect(GMCardSnapshot).toMatchSnapshot();
  });
});
