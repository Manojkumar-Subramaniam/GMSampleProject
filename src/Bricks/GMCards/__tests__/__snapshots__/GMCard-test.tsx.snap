// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GMCard Tests GMCard Snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "elevation": 1,
      "margin": 5,
      "opacity": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 5,
      "zIndex": 1,
    }
  }
/>
`;
